### ISSUE TITLE
weekly useful materials - 06/15 -

### ISSUE THREAD

**Issue by GENZITSU** [GENZITSU/UsefulMaterials#103]

**Comment by GENZITSU** [comment]
## LightGBMのearly_stoppingの仕様が変わったので、使用法を調べてみた

詳細に書かれている記事だけど要はcallback使いましょうということ

古い書き方
```python
# from https://qiita.com/c60evaporator/items/2b7a2820d575e212bcf4#%E3%82%AF%E3%83%AD%E3%82%B9%E3%83%90%E3%83%AA%E3%83%87%E3%83%BC%E3%82%B7%E3%83%A7%E3%83%B3%E5%AE%9F%E8%A1%8C%E6%99%82%E3%81%AE%E6%8C%99%E5%8B%95

verbose_eval = 0  # この数字を1にすると学習時のスコア推移がコマンドライン表示される
# early_stoppingを指定してLightGBM学習
gbm = lgb.train(params, dtrain,
                valid_sets=[dvalid],  # early_stoppingの評価用データ
                num_boost_round=10000,  # 最大学習サイクル数。early_stopping使用時は大きな値を入力
                early_stopping_rounds=10,
                verbose_eval=verbose_eval
                )
```

新しい書き方
```python
# from https://qiita.com/c60evaporator/items/2b7a2820d575e212bcf4#%E3%82%AF%E3%83%AD%E3%82%B9%E3%83%90%E3%83%AA%E3%83%87%E3%83%BC%E3%82%B7%E3%83%A7%E3%83%B3%E5%AE%9F%E8%A1%8C%E6%99%82%E3%81%AE%E6%8C%99%E5%8B%95

verbose_eval = 0  # この数字を1にすると学習時のスコア推移がコマンドライン表示される
# early_stoppingを指定してLightGBM学習
lgbr = lgb.LGBMRegressor(**params)
lgbr.fit(X_train, y_train, 
         eval_metric='rmse',  # early_stoppingの評価指標(学習用の'metric'パラメータにも同じ指標が自動入力される)
         eval_set=[(X_valid, y_valid)],
         callbacks=[lgb.early_stopping(stopping_rounds=10, 
                        verbose=True), # early_stopping用コールバック関数
                    lgb.log_evaluation(verbose_eval)] # コマンドライン出力用コールバック関数
         )
```

### コメント
順次移行していきたい

### 出典
[LightGBMのearly_stoppingの仕様が変わったので、使用法を調べてみた](https://qiita.com/c60evaporator/items/2b7a2820d575e212bcf4)

**Comment by GENZITSU** [comment]
## Loss Function Library - Keras & PyTorch

下記のようなロス関数の実装がまとめられているnotebook。

![スクリーンショット 2022-06-08 20 17 53](https://user-images.githubusercontent.com/19774756/172603813-81355db6-1214-4f19-a3e5-c8f38b2672af.png)

### コメント
ありがたい。

### 出典
[Loss Function Library - Keras & PyTorch](https://www.kaggle.com/code/bigironsphere/loss-function-library-keras-pytorch/notebook)

**Comment by GENZITSU** [comment]
## 2022年 コンピュータビジョン分野のトレンド

xpaper チャレンジの⽚岡さんがGTC2022年で発表した、CV分社の技術トレンドのまとめ。アーキテクチャーのトレンドや各CVタスクでのトレンドに加えて、学習方法やデータ生成についてのまとめもなされている。

### コメント
近年の動向をざっくり知るのにめちゃ助かる資料。

### 出典
[2022年 コンピュータビジョン分野のトレンド](https://twitter.com/home)

**Comment by GENZITSU** [comment]
## DeNA, MoT AI勉強会発表資料「顔認識と最近のArcFaceまわりと」 / Face Recognition & ArcFace papers

顔認識タスクの一般的な取り組み方や使用可能なデータセット、ArcFace以降のSOTAモデルたちをまとめているスライド。

![スクリーンショット 2022-06-08 20 43 44](https://user-images.githubusercontent.com/19774756/172608441-80b01e69-60fc-47bc-be39-8015237da6a0.png)

![スクリーンショット 2022-06-08 20 43 51](https://user-images.githubusercontent.com/19774756/172608454-675bc198-8fec-4222-8442-4f6bd1fa8144.png)

![スクリーンショット 2022-06-08 20 44 27](https://user-images.githubusercontent.com/19774756/172608461-beb0969e-85e1-4fb9-81ac-b8368543eb53.png)

![スクリーンショット 2022-06-08 20 44 37](https://user-images.githubusercontent.com/19774756/172608469-86315a2a-6595-4e88-a8e1-edd37fb6f123.png)

![スクリーンショット 2022-06-08 20 44 49](https://user-images.githubusercontent.com/19774756/172608477-4371b5ca-0670-4f25-baee-9bcad21019b0.png)

### コメント
SOTAの移り変わりが激しい分野なのでとても助かる。

### 出典
[DeNA, MoT AI勉強会発表資料「顔認識と最近のArcFaceまわりと」 / Face Recognition & ArcFace papers](https://speakerdeck.com/takarasawa_/face-recognition-and-arcface-papers)

**Comment by GENZITSU** [comment]
## [sithu31296](https://github.com/sithu31296) / [semantic-segmentation](https://github.com/sithu31296/semantic-segmentation)

semantice segmentationのSOTAモデルたちを実装しているライブラリ。
MMsegやsmpよりかは充実していないが、それらには実装されていないものもいくつかある。

![スクリーンショット 2022-06-08 21 23 43](https://user-images.githubusercontent.com/19774756/172615319-51fbb5b2-b848-458e-9e43-6f2db9ce175c.png)

![スクリーンショット 2022-06-08 21 23 49](https://user-images.githubusercontent.com/19774756/172615347-ff08f120-bcec-4b55-b92d-39428e57d99c.png)

### コメント
メモ

### 出典
 [sithu31296](https://github.com/sithu31296) / [semantic-segmentation](https://github.com/sithu31296/semantic-segmentation)

**Comment by GENZITSU** [comment]
## Word Tour: One-dimensional Word Embeddings via the Traveling Salesman Problem (NAACL 2022)

近くに埋め込まれた単語は近しい意味を持つという健在性を保ちながら単語ベクトルを一次元に圧縮する手法の提案。

![スクリーンショット 2022-06-09 21 42 37](https://user-images.githubusercontent.com/19774756/172850593-1fe8cabc-1768-4d21-be9d-15d650181a93.png)

![スクリーンショット 2022-06-09 21 42 47](https://user-images.githubusercontent.com/19774756/172850623-258b7dd2-a16a-4ae4-81e9-34640045a285.png)

![スクリーンショット 2022-06-09 21 42 56](https://user-images.githubusercontent.com/19774756/172850641-1a2fcfb1-ae3b-4aa0-ace1-d7397933a7c2.png)

![スクリーンショット 2022-06-09 21 43 23](https://user-images.githubusercontent.com/19774756/172850712-3d8ae050-50d6-4527-9369-9eeb17bf14c8.png)

![スクリーンショット 2022-06-09 21 43 37](https://user-images.githubusercontent.com/19774756/172850735-b72a904a-3ade-4aee-b5fe-9e7a31b39e30.png)

![スクリーンショット 2022-06-09 21 43 48](https://user-images.githubusercontent.com/19774756/172850751-75284010-b7ec-4402-b6f1-7051bceea30b.png)

![スクリーンショット 2022-06-09 21 43 55](https://user-images.githubusercontent.com/19774756/172850769-f76f0022-b7f9-465f-88fb-a27d6eee2cb0.png)

![スクリーンショット 2022-06-09 21 44 06](https://user-images.githubusercontent.com/19774756/172850797-d5163a33-008b-4d8b-a62e-e6b445563f46.png)

![スクリーンショット 2022-06-09 21 44 19](https://user-images.githubusercontent.com/19774756/172850824-48670972-4eee-438b-8159-817767677996.png)

### コメント
自然言語を一次元連続にするという全く新しい手法で、定性的にも定量的にも結果が出ているのがすごい。

### 出典
[Word Tour: One-dimensional Word Embeddings via the Traveling Salesman Problem (NAACL 2022)](https://github.com/GENZITSU/UsefulMaterials/issues/103)

**Comment by GENZITSU** [comment]
## Accelerate BERT inference with Hugging Face Transformers and AWS Inferentia

distilbert-baseをinf1インスタンス用にNeuron SDKで最適化することで、4並列でレイテンシー6msの推論を可能にするチュートリアル。4並列できる分普通のGPUよりも速い。

### コメント

inf1インスタンスは普通のgpuインスタンスよりも安いらしいので、使用を検討したいところ。

### 出典
[Accelerate BERT inference with Hugging Face Transformers and AWS Inferentia](https://huggingface.co/blog/bert-inferentia-sagemaker)

**Comment by GENZITSU** [comment]
## スマートデバイス用の高速なアノテーションツール / Fast Annotation system for SmarT devices

サイバーエージェントが開発しているスマートフォン向けアノテーションツールの紹介。

![スクリーンショット 2022-06-09 22 38 32](https://user-images.githubusercontent.com/19774756/172860958-d6cb7ef2-16da-47ee-a69d-cb2a54815562.png)

![スクリーンショット 2022-06-09 22 38 42](https://user-images.githubusercontent.com/19774756/172860981-9bd471dd-3ed3-4a32-b7b1-2990403e1c66.png)

![スクリーンショット 2022-06-09 22 38 49](https://user-images.githubusercontent.com/19774756/172860996-8773708c-b698-4ef1-9508-4b5444ce55c2.png)

![スクリーンショット 2022-06-09 22 38 59](https://user-images.githubusercontent.com/19774756/172861014-8bbb26bb-bc1b-4311-a054-ab7a0fc2ac41.png)

![スクリーンショット 2022-06-09 22 39 16](https://user-images.githubusercontent.com/19774756/172861047-07773c4b-e00e-4b2a-be10-db69e91809af.png)

### コメント
簡単なラベリングがPCではなくスマホでできると捗りそう。

### 出典
- [スマートデバイス用の高速なアノテーションツール / Fast Annotation system for SmarT devices](https://speakerdeck.com/shunyooo/fast-annotation-system-for-smart-devices)
- [CyberAgent](https://github.com/CyberAgent) / [fast-annotation-tool](https://github.com/CyberAgent/fast-annotation-tool)

**Comment by GENZITSU** [comment]
## PythonユーザによるRust入門

pythonとrustの代表的な違いが紹介されている。

### コメント
メモ

### 出典
[PythonユーザによるRust入門](https://speakerdeck.com/rmizuta3/pythonyuzaniyorurustru-men)

**Comment by GENZITSU** [comment]
## Hugging face t5-base-japaneseをTensor-RTで高速化してみた

TensorRT + fp16を用いることで、推論速度が1.25 → 0.26まで向上している

### コメント
メモ

### 出典
- [Hugging face t5-base-japaneseをTensor-RTで高速化してみた](https://zenn.dev/kwashizzz/articles/ml-tensorrt-t5)
- [公式チュートリアル](https://github.com/NVIDIA/TensorRT/blob/main/demo/HuggingFace/notebooks/t5.ipynb)
